Runtime/Core/CredentialManagement/CredentialsStore.cs

```diff
@@ -68,11 +68,16 @@ public RetriveAwsCredentialsResponse RetriveAwsCredentials(RetriveAwsCredentials
             {
                 ImmutableCredentials credentials = awsCredentials.GetCredentials();
 
-                return Response.Ok(new RetriveAwsCredentialsResponse()
+                if (_credentialProfileStoreChain.TryGetProfile(request.ProfileName, out var profile))
```

Comment:
We can reduce nesting by combining this if statement with the previous:
```
if (_credentialProfileStoreChain.TryGetAWSCredentials(request.ProfileName, out var awsCredentials)
    && _credentialProfileStoreChain.TryGetProfile(request.ProfileName, out var profile))
{
    ImmutableCredentials credentials = awsCredentials.GetCredentials();

    return Response.Ok(new RetriveAwsCredentialsResponse()
    {
        AccessKey = credentials.AccessKey,
        SecretKey = credentials.SecretKey,
        Region = profile.Region.SystemName,
    });
}
```

Reply:
Applied change